src/Pulsar.Client/Internal/HttpLookupService.fs

```diff
@@ -113,14 +104,13 @@ type internal HttpLookupService (config: PulsarClientConfiguration, _connectionP
             let randomServiceUri = config.ServiceAddresses[RandomGenerator.Next(0, config.ServiceAddresses.Length)]
             let topic: string = %topicName
             let topicRestPath = topic.Replace("persistent://","persistent/").Replace("non-persistent://","non-persistent/")
-            let! response = httpClient.GetStreamAsync (randomServiceUri.AbsoluteUri + $"lookup/v2/topic/%s{topicRestPath}")
-            let brokerResponse =
-                JsonSerializer.Deserialize<{|
-                    BrokerUrl : string
-                    BrokerUrlTls: string
-                    HttpUrl: string
-                    HttpUrlTls: string
-                |}>(response, jsonOptions)
+            let! brokerResponse =
+                randomServiceUri.AbsoluteUri + $"lookup/v2/topic/%s{topicRestPath}"
+                |> pulsarHttpClient.Get<{| BrokerUrl : string;
+                                           BrokerUrlTls: string;
+                                           HttpUrl: string;
+                                           HttpUrlTls: string |}>
+                |> Async.AwaitTask
```

Comment:
Since this method is not recursive/retriable, we don't need async here, Async.AwaitTask as well, and `GetBrokerInner` is not needed, this code can be moved to just `GetBroker`

Reply:
I removed async and Async.AwaitTask in `GetBrokerInner()`, and migrate this code to `GetBroker()` with `background{}` wrapper.